cpmpy/expressions/globalconstraints.py

```diff
@@ -289,7 +289,32 @@ def deepcopy(self, memodict={}):
 
     # TODO: value()
 
-# Numeric Global Constraints (with integer-valued return type)
+
+# syntax of the form 'if b then x == 9 else x == 0' is not supported
+# a little helper:
+class IfThenElse(GlobalConstraint):
+    def __init__(self, condition, if_true, if_false):
+        super().__init__("ite", [condition, if_true, if_false], is_bool=True)
+
+    def value(self):
+        condition, if_true, if_false = self.args
+        condition_val = argval(condition)
+        true_val = argval(if_true)
```

Comment:
Better to calculate the val of `true_val` and `false_val` lazily. I.e:
```python
if condition_val:
     return argval(if_true)
else:
     return argval(if_false)
```
As one of the arguments might be an incomplete function such as an element with weird indexes...